log-window.js

```diff
@@ -26,261 +26,87 @@ class EmeLogWindow {
-
-    // This has fields like an object.
-    if (obj.__fields__) {
-      format += ' ' + prettyPrint(obj.__fields__, indentation);
-    }
-
-    // This has a data array like an ArrayBufferView.
-    // TODO: Handle formatting for 16-bit and 32-bit values?
-    if (obj.__data__) {
-      const data = obj.__data__.slice();  // Make a copy
-      if (data.length == 0) {
-        format += '[]';
-      } else {
-        format += ' ' + '[\n';
-        while (data.length) {
-          const row = data.splice(0, 16);
-          format += indentation + '  ';
-          format += row.map(byteToHex).join(', ');
-          format += ',\n';
-        }
-        format += indentation + ']';
-      }
-    }
-    return format;
-  }
-
-  if (Array.isArray(obj)) {
-    // More compact representations for empty or 1-element arrays.
-    if (obj.length == 0) {
-      return '[]';
-    }
-    if (obj.length == 1) {
-      return `[${prettyPrint(obj[0], indentation)}]`;
-    }
-
-    let insides = '';
-    for (const entry of obj) {
-      insides += indentation + '  ';
-      insides += prettyPrint(entry, indentation + '  ') + ',\n';
-    }
-    return `[\n${insides}${indentation}]`;
-  }
-
-  if (obj.constructor == Object) {
-    const keys = Object.keys(obj);
-
-    // More compact representations for empty or 1-element objects.
-    if (keys.length == 0) {
-      return '{}';
-    }
-    if (keys.length == 1) {
-      return `{${keys[0]}: ${prettyPrint(obj[keys[0]], indentation)}}`;
-    }
-
-    let insides = '';
-    for (const key of keys) {
-      insides += indentation + '  ' + key + ': ';
-      insides += prettyPrint(obj[key], indentation + '  ') + ',\n';
-    }
-    return `{\n${insides}${indentation}}`;
-  }
-
-  if (typeof obj == 'string') {
-    return `"${obj}"`;
-  }
-
-  return obj.toString();
-}
 
 // NOTE: These APIs are not defined in our test environment, but should always
 // be present when this is run as a Chrome extension.
 if (chrome.runtime !== undefined) {
   /**
-   * Listens for messages from the content script to append a log item to the
-   * current frame and log file.
+   * When a window is closed, close the log window.
    */
-  chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
-    const log = request.log;
-    EmeLogWindow.instance.appendLog(log);
+  chrome.windows.onRemoved.addListener((windowId) => {
+    EmeLogWindow.instance.close(windowId);
   });
 
   /**
    * When the extension icon is clicked, open the log window if it doesn't exist,
    * and bring it to the front otherwise.
    */
-  chrome.browserAction.onClicked.addListener((tab) => {
+  chrome.action.onClicked.addListener(async (tab) => {
     EmeLogWindow.instance.open();
   });
+
+  chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
+    if (message.type === 'EME_LOGGER_CLEAR') {
+      EmeLogWindow.instance.clear();
+    }
+    else if (message.type === 'EME_LOGGER_GET_TEXT_LOGS') {
```

Comment:
style nit: Please move to previous line

TODO: I need to add a style linter.  :-)

Reply:
Done